### DISCUSSION TITLE
non-concentric sample and detector circles.

### DISCUSSION THREAD

**Discussion by brussel13** [dkriegner/xrayutilities#198] [General]
[diffractometer.pdf](https://github.com/user-attachments/files/17461101/diffractometer.pdf)

Greetings.  We are preparing for an experiment at 1-ID of the APS.  The diffractometer is configured as illustrated in the attached pdf.  The detectors shown under the red X are not going to be in place.  We will use the detector positioning in the green circle on the left.  It's motion is not concentric with the center of the sample goniometer (circled in green on the right).  It is centered on the direct beam significantly downstream of the sample.  There is also a radius from the direct beam to specify.
Is there an obvious way to specify this detector circle and radius in the Qconversion class?

Thanks,
Ross

**Comment by dkriegner** [comment]
Combining translations and rotations in the right order this should be straightforward. If a translation is inserted in the goniometer definition it shifts the rotation center of subsequent circles (this makes the calculations internally a bit slower but is usually not noticeable). So if i understand the geometry correctly then you might need only one translation and one rotation for this.
